krabs/krabs/wstring_convert.hpp

```diff
@@ -0,0 +1,31 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <windows.h>
+
+namespace krabs {
+
+	/** <summary>
+	  * Converts std::wstring argument to std::string using UTF-8 codepage
+	  * Returns empty string if translation fails or input string is empty
+	  * </summary>
+	  */
+	inline std::string from_wstring(const std::wstring& wstr, UINT codePage = CP_UTF8)
+	{
+		if (wstr.empty())
+			return {};
+
+		const auto requiredLen = WideCharToMultiByte(codePage, 0, wstr.data(), static_cast<int>(wstr.size()), 
+			nullptr, 0, nullptr, nullptr);
+		if (0 == requiredLen)
+			return {};
+
+		std::string result;
+		result.resize(requiredLen);
```

Comment:
optional: consider combining the allocation and the resize: `std::string result(requiredLen, 0);`